packages/query/src/block-processor.ts

```diff
@@ -100,8 +101,8 @@ export class BlockProcessor implements BlockProcessorInterface {
 
   // If sync() is called multiple times concurrently, they'll all wait for
   // the same promise rather than each starting their own sync process.
-  public sync = (): Promise<void> =>
-    (this.syncPromise ??= backOff(() => this.syncAndStore(), {
+  public sync = (genesisBlock?: JsonObject): Promise<void> =>
```

Comment:
Think it would be easier for the consumer to pass this into the class instance: https://github.com/prax-wallet/web/pull/165/files#diff-3483cabb881c2f2296418077d1f80daeadfb36525092b964211b678724314192R110-R111 and have it as a private property

Reply:
this is a great suggestion; it's much cleaner to pass this into the class instance